build/tasks/debug.js

```diff
@@ -61,7 +61,7 @@ module.exports = function (grunt) {
         if (func) {
           // FunctionExpression.BlockStatement.body(Array)
           var funcBody = func.body.body;
-          esprima.parse('blocks.debug && blocks.debug.checkArgs(' + toValueString(data) + ', Array.prototype.slice.call(arguments), {})').body.forEach(function (chunk) {
+          esprima.parse('blocks.debug && blocks.debug.checkArgs(__DEBUG_METHOD, Array.prototype.slice.call(arguments), {}); var __DEBUG_METHOD = ' + toValueString(data) + ';').body.forEach(function (chunk) {
```

Comment:
You could do:

This is pseudo code. I haven't tested it. I hope you get the idea.
```
func.body.body.unshift.apply(
  func.body.body,
  esprima.parse('var __DEBUG_METHOD = ' + toValueString(data) + ';' blocks.debug && blocks.debug.checkArgs(__DEBUG_METHOD, Array.prototype.slice.call(arguments), {});').body
);
```

Regarding .jshintc - I leaving this to you to decide.

P.S. When you are ready with the changes you could go ahead and merge this yourself.